network/requestTracker.go

```diff
@@ -532,7 +545,7 @@ func (rt *RequestTracker) ServeHTTP(response http.ResponseWriter, request *http.
 	rateLimitedRemoteHost := (!rt.config.DisableLocalhostConnectionRateLimit) || (!isLocalhost(trackedRequest.remoteHost))
 	connectionLimitEnabled := rt.config.ConnectionsRateLimitingWindowSeconds > 0 && rt.config.ConnectionsRateLimitingCount > 0
 
-	if originConnections > rt.config.ConnectionsRateLimitingCount && connectionLimitEnabled && rateLimitedRemoteHost {
+	if originConnections > rt.config.ConnectionsRateLimitingCount && connectionLimitEnabled && rateLimitedRemoteHost && !healthServiceInvocation {
```

Comment:
Good question!
 
In this case it's conservatively done, as health status invocation should not be subject to normal traffic rules (as used for internal balancing / health status) and where you regardless of custom configuration would not want it subject to be throttled, as that would risk flagging an otherwise healthy server as unhealthy.
 
With the defaults (60 connections over 1 time window) I consider it highly unlikely for this unhealthy status to happen for anyone using a default configuration, however in the case of custom configurations (of which we do not know the full set) there is a possibility that someone may be throttling more aggressively, and in which case you would still want the health service to be exempt.